app/components/satis/dropdown/component_controller.js

```diff
@@ -92,16 +92,18 @@ export default class extends ApplicationController {
       this.getChainToElement()?.addEventListener("change", this.boundChainToChanged)
     }
 
-    this.refreshSelectionFromServer().then((changed) => {
-      this.filterResultsChainTo()
-      this.setHiddenSelect()
-
-      if (!this.hiddenSelectTarget.getAttribute("data-reflex")) {
-        let event = new Event("change")
-        event.detail = { src: "satis-dropdown" }
-        this.hiddenSelectTarget.dispatchEvent(event)
-      }
-    })
+    if (this.hiddenSelectTarget.selectedOptions[0].value) {
```

Comment:
So we basically have an ID, or multiple selectedOptions?
If so, are we only fetching those options instead of all?

Reply:
In this this.hiddenSelectTarget.selectedOptions[0].value we are getting selected option only, if we have selected value for dropdown we are fetching all values here from server, so just restricted that if we have selected ones.
Thanks!

Reply:
looks good but if selectedOptions is empty then it will throw index out of bound error, may be need to add extra check to make sure selectedOptions length is greater than 0

Reply:
yes sure, added this safe check!
Thank you for review!